File: src/core/jupiter/core/apps/life_plan/note-sections.ts
import { RecurringTaskPeriod } from "@jupiter/webapi-client";

import { comparePeriods } from "#/core/common/recurring-task-period";

export function periodAllowsAspectsInNote(
  period: RecurringTaskPeriod,
): boolean {
  return comparePeriods(period, RecurringTaskPeriod.MONTHLY) >= 0;
}

export function periodAllowsGoalsInNote(period: RecurringTaskPeriod): boolean {
  return comparePeriods(period, RecurringTaskPeriod.QUARTERLY) >= 0;
}

File: src/core/jupiter/core/apps/life_plan/note_sections.py
"""Rules for which life plan sections a note of a certain period can have."""

from jupiter.core.common.recurring_task_period import RecurringTaskPeriod


def period_allows_aspects_in_note(period: RecurringTaskPeriod) -> bool:
    """Whether a note for this period can have a section for each aspect."""
    return period >= RecurringTaskPeriod.MONTHLY


def period_allows_goals_in_note(period: RecurringTaskPeriod) -> bool:
    """Whether a note for this period can have a section for each goal."""
    return period >= RecurringTaskPeriod.QUARTERLY
